mqtt-sn.c

```diff
@@ -841,7 +847,11 @@ void* mqtt_sn_wait_for(uint8_t type, int sock)
             if (packet) {
                 switch(packet[1]) {
                     case MQTT_SN_TYPE_PUBLISH:
-                        mqtt_sn_print_publish_packet((publish_packet_t *)packet);
+                        rc = mqtt_sn_print_publish_packet((publish_packet_t *)packet);
+                        if(type == packet[1]){
+                        	return packet;
+                        }
+                        mqtt_sn_send_puback(sock,(publish_packet_t *)packet,rc);
```

Comment:
Puback is sent in https://github.com/njh/mqtt-sn-tools/blob/ff65892a1c6755cc9eb2762bc3099e43a4ce1f8b/mqtt-sn-sub.c#L260

I am trying to remember why `mqtt_sn_print_publish_packet()` is inside `mqtt_sn_wait_for()`

But I guess that is the same reason why you are missing some PUBACKs.

Reply:
Ahh, it was because of issue #4